File: server/src/MMO-Idle.Domain/Entities/CharacterSkill.cs
using MMOIdle.Domain.Entities;
using MMOIdle.Domain.Enums;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MMOIdle.Domain.Entities
{
    public class CharacterSkill
    {
        public long Id { get; set; }
        public Guid CharacterId { get; set; }
        public LifeSkillType SkillType { get; set; }
        public int Level { get; set; } = 1;
        public int Experience { get; set; } = 0;
        public DateTime LastUpdated { get; set; } = DateTime.UtcNow;
        public virtual Character? Character { get; set; }
    }

}

File: client/src/models/CharacterSkill.ts
import { LifeSkillTypeEnum } from "../enums/LifeSkillTypeEnum";

export interface CharacterSkill {
    id: string;
    name: string;
    skillType: LifeSkillTypeEnum;
    level: number;
    experience: number;
    lastUpdated: Date;
} 